### PR TITLE
feat: Package the software with nix using devenv.nix

### PR DESCRIPTION
**Pull request by k3yss**
This pull request packages `clang-pong` software using nix allowing always reproducible  build using `devenv.lock` file.

### CODE REVIEW

**Comment by gergoerdi** [comment]
Thank you!

Is there a way for me to try this out on a non-NixOS Linux distribution?

**Comment by k3yss** [comment]
@gergoerdi Absolutely! I packaged this on Arch linux. 

You can install [devenv](https://devenv.sh/getting-started/) and it would work like any other binary that can be used to invoke this.

Alternatively you can also explore [nix-shell](https://nixos.wiki/wiki/Development_environment_with_nix-shell) for `.nix` files.

As this script is only for `devenv` so I would recommend you install the former.

**Comment by gergoerdi** [comment]
OK I got sidetracked on a wild goose chase involving Clashilator link errors, to upgrade to Clash 1.8 / GHC 9.4. With that out of the way, I can look at this now.

Looking at the files, it is unclear to me what actual information they hold vs. just being autogenerated stock files.  If another `devenv` user were to come around, would they not be able to recreate the exact same files using some `devenv`-generator that I guess must exist?

In other words, what is the value provided by having these files in the repo supposed to be, compared to you just having them locally and adding them to your `.git/info/exclude`?
